cmd/cluster-agent/app/app.go

```diff
@@ -247,6 +249,20 @@ func start(cmd *cobra.Command, args []string) error {
 		}()
 	}
 
+	// Generate and persist a cluster ID
+	// this must be a UUID, and ideally be stable for the lifetime of a cluster
+	// so we store it in a configmap that we try and read before generating a new one.
+	cacheClusterIDKey := cache.BuildAgentKey(clustername.ClusterIDCacheKey)
```

Comment:
Is it possible to have it in the cache at this point of the code? 🤔

Reply:
good catch, it won't be there at this point, removing.